### PR TITLE
Add FAQ section for creating reproducible results

### PR DESCRIPTION
**Pull request by td2014**
Based on the outcome of a recent github conversation about reproducibility, a section of python code was developed that illustrate how to get reproducible results from run-to-run, which can help debugging and comparisons during development.  I summarized the most recent code example from the github thread and added some commentary text in the FAQ to explain it.  The discussion thread is here:  https://github.com/fchollet/keras/issues/2280#issuecomment-306959926

Thanks.

### CODE REVIEW

File: docs/templates/getting-started/faq.md [hunk 1]
```diff
@@ -454,3 +455,29 @@ It contains the following fields:
...
+os.environ['PYTHONHASHSEED'] = '0'
+np.random.seed(42)
+rn.seed(12345)
+#single thread
+session_conf = tf.ConfigProto(
+intra_op_parallelism_threads=1,
+inter_op_parallelism_threads=1)
+
+from keras import backend as K
+tf.set_random_seed(1234)
```
Comment (fchollet) [inline docs/templates/getting-started/faq.md]: You are setting 4 different random seeds. There should be detailed comments explaining what each seed is for, and whether the user needs it.

File: docs/templates/getting-started/faq.md [hunk 2]
```diff
@@ -454,3 +455,29 @@ It contains the following fields:
...
+
+```python
+import numpy as np
+import tensorflow as tf
+import random as rn
+import os
+os.environ['PYTHONHASHSEED'] = '0'
+np.random.seed(42)
+rn.seed(12345)
+#single thread
```
Comment (fchollet) [inline docs/templates/getting-started/faq.md]: Need a more detailed comment. Style-wise, a comment should start with a space.

File: docs/templates/getting-started/faq.md [hunk 3]
```diff
@@ -454,3 +455,29 @@ It contains the following fields:
...
+session_conf = tf.ConfigProto(
+intra_op_parallelism_threads=1,
+inter_op_parallelism_threads=1)
```
Comment (fchollet) [inline docs/templates/getting-started/faq.md]: Fix PEP8 style

File: docs/templates/getting-started/faq.md (hunk 2 above)
Comment (td2014) [inline docs/templates/getting-started/faq.md]: Cleaned up comment per style and added additional details.

File: docs/templates/getting-started/faq.md (hunk 3 above)
Comment (td2014) [inline docs/templates/getting-started/faq.md]: Changed function call to PEP8 style (made a single line).  Hopefully this is okay.

File: docs/templates/getting-started/faq.md (hunk 1 above)
Comment (td2014) [inline docs/templates/getting-started/faq.md]: Provided additional details and some references.

**Comment by fchollet** [comment]
I don't see the changes. Additionally, you have committed the `faq.md` file in `sources`, which is autogenerated. Please revert.

**Comment by td2014** [comment]
Reverted (removed autogenerated faq.md under sources).  Moved changes to faq.md under templates.

**Comment by fchollet** [comment]
LGTM. Please resolve the merge conflicts (should be minor).

**Comment by td2014** [comment]
Merge conflict resolved (required minor edit) and successfully submitted.  Thanks.
